### PR TITLE
Feature/artifact name

### PR DESCRIPTION
**Pull request by teone**
[SDRAN-1480] Adding ArtifactName property so that artifacts can be named differently than the model itself
    
This is needed as in some cases we have different versions of the model coexisting at the same time, eg `aether-2.x` and `aether-4.0.x`.
It is not enough to set a different version of the artifact as we want to be able to generate a Docker container with tag `latest` for both of them.

**NOTE** That in order for this to work with `onos-config` you need to make sure the `Device` you create in `onos-topo` use the correct model name (it does not contain the version anymore), eg:

```
onos topo create entity devicesim-1 -a onos.topo.Configurable='{"address":"devicesim1-device-simulator:11161","version":"1.0.0","type":"devicesim","timeout":"30000000000"}' -a onos.topo.TLSOptions='{"insecure":true,"plain":true}' -a onos.topo.Asset='{"name":"Devicesim1"}' -a onos.topo.MastershipState='{}' -k devicesim
```

### CODE REVIEW

File: models/testdevice-2.0.x/openapi.yaml [hunk 1]
```diff
@@ -288,11 +288,11 @@ info:
     name: LicenseRef-ONF-Member-1.0
     url: https://opennetworking.org/wp-content/uploads/2020/06/ONF-Member-Only-Software-License-v1.0.pdf
   termsOfService: https://opennetworking.org/wp-content/uploads/2019/02/ONF-Licensing-and-IPR-FAQ-2020-06.pdf
-  title: testdevice-2.0.x-2.0.0
+  title: testdevice-2.0.0
   version: 2.0.0
 openapi: 3.0.0
 paths:
-  /testdevice-2.0.x/v2.0.0/{target}/cont1a:
+  /testdevice/v2.0.0/{target}/cont1a:
```
Comment (SeanCondon) [inline models/testdevice-2.0.x/openapi.yaml:295]: nice - this is much better

File: models/testdevice-2.0.x/metadata.yaml [hunk 2]
```diff
@@ -1,6 +1,7 @@
-name: testdevice-2.0.x
+name: testdevice
 version: 2.0.0
-goPackage: github.com/onosproject/config-models/models/testdevice-2.0.x
+artifactName: testdevice-2.0.x
```
Comment (SeanCondon) [inline models/testdevice-2.0.x/metadata.yaml:3]: perfect

**Review by SeanCondon — APPROVED** [review]

**Review by tomikazi — APPROVED** [review]
Let's also see what Sean thinks.

File: models/testdevice-2.0.x/openapi.yaml (hunk 1 above)
Comment (tomikazi) [reply models/testdevice-2.0.x/openapi.yaml:295]: Agreed!

**Comment by teone** [comment]
retest this please

**Pull request merged**